### DISCUSSION TITLE
Different zip and checksum from Github vs. ImageMagick.org Downloads?

### DISCUSSION THREAD

**Discussion by tdprogramming** [ImageMagick/ImageMagick#3138] [Development]
I've been working on a build pipeline which is building ImageMagick from source for an Amazon Linux based Docker Image.

As I'm downloading the source zip and building it as a build step, I also added a step to check the checksums of the downloaded zip.

For the Zip at https://github.com/ImageMagick/ImageMagick/archive/7.0.10-58.zip

The sha 256 checksum is:

88cfad2993acab65d96ba42c7d1f659403d2d3534eb88fb4535cfe84c74a9cd1

For the zip at https://download.imagemagick.org/ImageMagick/download/ImageMagick-7.0.10-58.zip

The sha 256 checksum is:

6a6877263f51de8d3d7011314bae66cf732f542ad0ee447cb3fe6518446030a1

(this checksum is confirmed in the digest.rdf also available from ImageMagick.org).

They also seem to compile differently - the zip from GitHub seems to require different dependencies for Webp and fails in my current pipeline, whereas the ImageMagick.org zip builds fine.

To help me maintain this system, and to reassure me that both zips are secure - what are the differences? Is the zip at Github for building the AppImage whereas the one at ImageMagick.org is for a traditional make & install? Why are they different? Also, if the difference is intentional, it would be great to also have checksums published with the releases on GitHub so that we can verify them.

I appreciate it's a nitpick but if anyone knows the answers, please do let me know!

Many thanks to all who work on this wonderful open source product.

**Comment by urban-warrior** [comment]
The distributions are essentially the same but have different automake/autoconf releases.  Github uses automake 1.16.2 whereas ImageMagick Studio, 1.16.1.  Since there are small difference in the build, you would can expect different checksums.  ImageMagick, in addition to providing a checksum, also provides a digital certificate to verify trust in the ImageMagick builds.

The builds are not intentionally different, they are just slightly different build environments.  It is not a priority for us to make them exactly the same.
